src/Microsoft.AspNet.Http/Authentication/DefaultAuthenticationManager.cs

```diff
@@ -0,0 +1,147 @@
+            {
+                throw new InvalidOperationException("The following authentication scheme was not accepted: " + authenticationScheme);
+            }
+
+            return new AuthenticationResult(authenticateContext.Principal,
+                new AuthenticationProperties(authenticateContext.Properties),
+                new AuthenticationDescription(authenticateContext.Description));
+        }
+
+        public override async Task<AuthenticationResult> AuthenticateAsync([NotNull] string authenticationScheme)
+        {
+            var handler = HttpAuthenticationFeature.Handler;
+
+            var authenticateContext = new AuthenticateContext(authenticationScheme);
+            if (handler != null)
+            {
+                await handler.AuthenticateAsync(authenticateContext);
+            }
+
+            // Verify all types ack'd
+            if (!authenticateContext.Accepted)
+            {
+                throw new InvalidOperationException("The following authentication scheme was not accepted: " + authenticationScheme);
+            }
+
+            return new AuthenticationResult(authenticateContext.Principal,
+                new AuthenticationProperties(authenticateContext.Properties),
+                new AuthenticationDescription(authenticateContext.Description));
+        }
+
+        public override void Challenge(AuthenticationProperties properties, string authenticationScheme)
+        {
+            HttpResponseFeature.StatusCode = 401;
+            var handler = HttpAuthenticationFeature.Handler;
+
+            var challengeContext = new ChallengeContext(authenticationScheme, properties == null ? null : properties.Items);
+            if (handler != null)
+            {
+                handler.Challenge(challengeContext);
+            }
+
+            // The default Challenge with no scheme is always accepted
+            if (!challengeContext.Accepted && !string.IsNullOrEmpty(authenticationScheme))
+            {
+                throw new InvalidOperationException("The following authentication type was not accepted: " + authenticationScheme);
+            }
+        }
+
+        public override void SignIn([NotNull] string authenticationScheme, [NotNull] ClaimsPrincipal principal, AuthenticationProperties properties)
+        {
+            var handler = HttpAuthenticationFeature.Handler;
+
+            var signInContext = new SignInContext(authenticationScheme, principal, properties?.Items);
+            if (handler != null)
+            {
+                handler.SignIn(signInContext);
+            }
+
+            // Verify all types ack'd
+            if (!signInContext.Accepted)
+            {
+                throw new InvalidOperationException("The following authentication scheme was not accepted: " + authenticationScheme);
+            }
+        }
+
+        public override void SignOut(string authenticationScheme, AuthenticationProperties properties)
```

Comment:
I've seen enough examples of scheme being used without properties that I think it's more convenient to keep them separate.